src/types/graphql.d.ts

```diff
@@ -0,0 +1,29 @@
+import createLoaders from "../lib/loaders"
+
+export interface RootValues {
+  /** Optionally you can include an access token for a logged in user */
+  accessToken: string | undefined
+  /** Optionally you can include a user token for a logged in user */
+  userID: string | undefined
+  /** Optionally you should include a timezone for the user */
+  defaultTimezone: string | undefined
+
+  /** The schema used by the internal exchange graphql engine */
+  exchangeSchema: GraphQLSchema
+}
+
+/**
+ * The root values we pass into the GraphQL query resolver,
+ * contains optional user information, and all of our data-loaders.
+ * */
+export type RootValue = RootValues & ReturnType<typeof createLoaders>
+
+declare module "graphql/type" {
+  interface GraphQLResolveInfo {
+    /**
+     * Where we store user/data-loader context, yes there's a context
+     * object but without a big refactor it's not happening soon.
+     */
+    readonly rootValue: RootValue
+  }
+}
```

Comment:
worth noting this is `d.ts` file - which get special rules. They're called ambient declaration files, and can't have any "runtime" JS code inside them. 

This one exists to inject extra code into the `graphql/types` module ( which resolves to `@types/graphql/types.d.ts` )